rust/src/router/webrtc_transport.rs

```diff
@@ -80,33 +81,50 @@ impl TryFrom<Vec<TransportListenIp>> for TransportListenIps {
     }
 }
 
-/// [`WebRtcTransport`] options.
+/// How [`WebRtcTransport`] should listen on interfaces.
 ///
 /// # Notes on usage
 /// * Do not use "0.0.0.0" into `listen_ips`. Values in `listen_ips` must be specific bindable IPs
 ///   on the host.
 /// * If you use "0.0.0.0" or "::" into `listen_ips`, then you need to also provide `announced_ip`
 ///   in the corresponding entry in `listen_ips`.
+#[derive(Debug, Clone)]
+pub enum WebRtcTransportListen {
+    /// Listen on individual IP/port combinations specific to this transport.
+    Individual {
+        /// Listening IP address or addresses in order of preference (first one is the preferred one).
+        listen_ips: TransportListenIps,
+        /// Fixed port to listen on instead of selecting automatically from Worker's port range.
+        port: Option<u16>,
+        /// Listen in UDP. Default true.
+        enable_udp: bool,
+        /// Listen in TCP.
+        /// Default false.
+        enable_tcp: bool,
+        /// Prefer UDP.
+        /// Default false.
+        prefer_udp: bool,
+        /// Prefer TCP.
+        /// Default false.
+        prefer_tcp: bool,
```

Comment:
@ibc I meant to ask earlier, but want to confirm whether this separation is actually correct or some of those flags are still applicable to WebRtcServer scenario.

Reply:
OF COURSE! And they work as intended. See:

- https://github.com/versatica/mediasoup/blob/v3/worker/src/RTC/Router.cpp#L270
- https://github.com/versatica/mediasoup/blob/v3/worker/src/RTC/WebRtcServer.cpp#L283

Reply:
I'll have to partially revert this change then so they are usable again, thanks!

Reply:
Sorry, I didn't realize you removed them. The thing is that, based on those flags, resulting ICE candidates may vary (based on whether the WebRtcServer was provided or not with UDP and TCP ports).

Reply:
Yes, makes sense. They're not removed, they're just not specifiable with WebRtcServer specifically. I'll bring them back.

Reply:
Just to clarify, nothing changed at API level. Those flags just exist and still exist when you call `router.createWebRtcTransport()` despite you pass a `WebRtcServer` or not into the options.